.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+---
+repos:
+  - repo: "https://github.com/astral-sh/ruff-pre-commit"
+    # Ruff version.
+    rev: "v0.6.1"
+    hooks:
+      # Run the linter.
+      - id: "ruff"
+        exclude: ".*_pb2.*"
+      # Run the formatter.
+      - id: "ruff-format"
+        exclude: ".*_pb2.*"
+  - repo: "https://github.com/adrienverge/yamllint"
+    rev: "v1.35.1"
+    hooks:
+      - id: "yamllint"
+  - repo: "https://github.com/RobertCraigie/pyright-python"
```

Comment:
I also swapped from mypy to pyright - pyright is the one that vscode uses, and it's got some performance and typing benefits over mypy. There's a comparison here: https://github.com/microsoft/pyright/blob/main/docs/mypy-comparison.md